File: goosip_SpringAPI/src/main/java/com/chatsapp/goosip/model/user/UserStory.java
package com.chatsapp.goosip.model.user;

import java.util.ArrayList;

public class UserStory {

    private String uid;
    private String name;
    private String ProfileImg;

    public UserStory(String uid, String name, String profileImg, ArrayList<Story> stories) {
        this.uid = uid;
        this.name = name;
        this.ProfileImg = profileImg;
        this.stories = stories;
    }

    private ArrayList<Story> stories;

    public String getUid() {
        return uid;
    }

    public void setUid(String uid) {
        this.uid = uid;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public ArrayList<Story> getStories() {
        return stories;
    }

    public void setStories(ArrayList<Story> stories) {
        this.stories = stories;
    }

    public UserStory() {
    }

    public String getProfileImg() {
        return ProfileImg;
    }

    public void setProfileImg(String profileImg) {
        this.ProfileImg = profileImg;
    }
}

File: goosip_androidApp/app/src/main/java/com/android/chatsapp/model/UserStory.kt
package com.android.chatsapp.model

import android.os.Parcel
import android.os.Parcelable

data class UserStory(
    var uid: String?,
    var name: String?,
    var profileImg: String?,
    var stories: ArrayList<Story>
) : Parcelable {
    constructor(parcel: Parcel) : this(
        parcel.readString(),
        parcel.readString(),
        parcel.readString(),
        parcel.createTypedArrayList(Story.CREATOR) ?: ArrayList()
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(uid)
        parcel.writeString(name)
        parcel.writeString(profileImg)
        parcel.writeTypedList(stories)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<UserStory> {
        override fun createFromParcel(parcel: Parcel): UserStory {
            return UserStory(parcel)
        }

        override fun newArray(size: Int): Array<UserStory?> {
            return arrayOfNulls(size)
        }
    }
}
